goth_tests/domain/ya-provider/test_provider_multi_activity.py

```diff
@@ -159,3 +159,184 @@ async def test_provider_single_simultaneous_activity(
 
         await requestor.terminate_agreement(agreement_id, None)
         await provider.wait_for_agreement_terminated()
+
+
+@pytest.mark.asyncio
+async def test_provider_renegotiate_proposal(
+    common_assets: Path,
+    config_overrides: List[Override],
+    log_dir: Path,
+):
+    """Tests providers' ability of renegotiating previously rejected proposal."""
+
+    nodes = [
+        {"name": "requestor-1", "type": "Requestor"},
+        {"name": "requestor-2", "type": "Requestor"},
+        {"name": "provider-1", "type": "VM-Wasm-Provider", "use-proxy": True},
+    ]
+    config_overrides.append(("nodes", nodes))
+
+    runner, config = _create_runner(common_assets, config_overrides, log_dir)
+
+    async with runner(config.containers):
+        requestor1, requestor2 = runner.get_probes(probe_type=RequestorProbe)
+        providers = runner.get_probes(probe_type=ProviderProbe)
+        assert providers
```

Comment:
I've added this line as @zakaprov did in above tests, and indeed `providers` was empty list